mlflow/pipelines/utils/execution.py

```diff
@@ -400,6 +400,25 @@ def _create_makefile(pipeline_root_path, execution_directory_path, template) ->
 
```

Comment:
Apologies for not asking earlier, but is `_create_makefile` definitely the right place to implement this logic? And more broadly, `_get_or_create_execution_dir`? This is creating files within the source repository and not the execution directory, correct?

Reply:
`_create_makefile ` will run before running a step. Same for `_get_or_create_execution_directory`.
I put it in `_create_makefile ` because there's already a template check inside the method and it only makes sense to check step files for `regression/v1`. The step files are required by the specific make file - that another reason we are creating them together.

Maybe rename the method to "_create_makefile_and_steps"?

A better approach will be create a layer for project integrity check during `new Pipeline(x)`.